extensions/hide_uncommon_channels.c

```diff
@@ -24,7 +24,8 @@ mapi_hfn_list_av1 huc_hfnlist[] = {
 DECLARE_MODULE_AV2(hide_uncommon_channels, NULL, NULL, NULL, NULL, huc_hfnlist, NULL, NULL, hide_desc);
 
 static void
-h_huc_doing_whois_channel_visibility(hook_data_client *hdata)
+h_huc_doing_whois_channel_visibility(void *data_)
 {
-	hdata->approved = ((PubChannel(hdata->chptr) && !IsInvisible(hdata->target)) || IsMember((hdata->client), (hdata->chptr)));
+	hook_data_channel_visibility *data = data_;
+	data->approved = data->approved && (!IsInvisible(data->targms->client_p) || data->clientms != NULL);
```

Comment:
To go with the struct change:

```suggestion
	data->approved = data->approved && (!IsInvisible(data->targetms->client_p) || data->clientms != NULL);
```

Alternately, if we add the target to the hook struct:

```suggestion
	data->approved = data->approved && (!IsInvisible(data->target) || data->clientms != NULL);
```